### ISSUE TITLE
Metaliminal Storms Endpoint

### ISSUE THREAD

**Issue by november781** [esi/esi-issues#1224]
# Feature Request

I would like to see an endpoint that would allow players to pull the current locations of the new metaliminal storms

## Use case

I actually primarily want to use this to allow the creation of a new eden weather service without having to direct the combined might of Signal Cartel to nullsec to find them, and I suppose it could be used by brave players to find the storms either for PVP or PVE activities 

## Authentication

A public route, no auth required.

## Example return

Propose what an example return might look like. E.g.:

```json
[
  {
    "Storm": "Strong Metaliminal Gamma Ray Storm",
    "System_id": [
                           30003271
                         ],
  },
  {
    "Storm": "Weak Metaliminal Gamma Ray Storm",
    "System_id" : [
                           30003274,
                           30003269
                          ]
  } ,
 ... other 6 storms
]
```

# Checklist

Check all boxes that apply to this issue:

- [x] Feature request description is provided
- [x] Use case exists
- [x] Feature requires a new route
- [ ] Feature adds data to existing route
- [ ] Feature requires new auth scope
- [ ] Feature can reuse existing scope
- [x] Feature does not require auth
- [ ] Meta feature, applies to all routes

**Comment by Blacksmoke16** [comment]
Aren't these storms just part of an event?  I.e. temporary.

**Comment by november781** [comment]
Going off of the [forum post from Fozzie ](https://forums.eveonline.com/t/coming-soon-metaliminal-storms-in-nullsec/258511), the storm types might be, but the presence of some storms is not.

I could be wrong.

**Comment by guiguilechat** [comment]
IMO it would be better to have

1. a universe/weather/{weather_id} path that returns the information about a weather id (its effect, name among other things)
2. universe/weather/weathers path that returns a list of the couples [systemid:weatherid] containing all the systems affected by a weather, and the id of the weather affecting that system. Actually it may be better to have a WEATHERINFO structure containing the weather id, the date since the weather affects that system, the weather strength . Not that there is no reason for weathers to not overlap, especially if later weather system integrates invasion, incursions, trig/eden victory, so there is no reason to assume the id of a system to be of limited occurence here.

**Comment by november781** [comment]
> 2\. universe/weather/weathers path that returns a list of the couples [systemid:weatherid] containing all the systems affected by a weather, and the id of the weather affecting that system. Actually it may be better to have a WEATHERINFO structure containing the weather id, the date since the weather affects that system, the weather strength . Not that there is no reason for weathers to not overlap, especially if later weather system integrates invasion, incursions, trig/eden victory, so there is no reason to assume the id of a system to be of limited occurence here.

Thats a good point, I was mostly just wanting to get a discussion on this going ahead of the public tests since there is a not-insignificant number of players who have a similar desire for ESI based tracking of these

**Comment by alexbaileyuk** [comment]
I would definitely second this. I can think of many apps and apis that would benefit from this.

**Comment by Risingson** [comment]
A new endpoint for dynamic solarsystem effects would be a good place to add all incursions, invasions and weather solarsystems.

**Comment by ccee2abf-fb28-497b-88ca-ae3643da4a8f** [comment]
Since the storms are buffs and debuffs having a endpoint that lists all the possible storms with its dogma specific information would be cool. Maybe turn it into something generic so we can pull information about abyssal and other system buffs as well.

**Comment by november781** [comment]
> 
> 
> A new endpoint for dynamic solarsystem effects would be a good place to add all incursions, invasions and weather solarsystems.



> 
> 
> Since the storms are buffs and debuffs having a endpoint that lists all the possible storms with its dogma specific information would be cool. Maybe turn it into something generic so we can pull information about abyssal and other system buffs as well.

so would something like `universe/weather/{weather_id}` that returns something like

```
[
  {
    "Weather": "Strong Metaliminal Gamma Ray Storm",
    "System_id": [
                           30003271
                         ],
    "Effects": {
                   "Explosive Resists": "-25%",
                   "Shield HP": "+25%"
                   .......
                  }
  },
  {
    "Weather": "Weak Metaliminal Gamma Ray Storm",
    "System_id" : [
                           30003274,
                           30003269
                          ],
   "Effects":{
                  .......
                 }
  } ,
  {
    "Weather": "Sansha Vanguard",
    "System_id": [
                        ......
                        ],
    "Effects": {
                     .............
                   }
   }
]
```
fit those use-cases

**Comment by guiguilechat** [comment]
no, you want to split the description of the weather and the relation between systems and weathers in separate paths

**Comment by dansi21** [comment]
Any development towards this?

**Comment by Blacksmoke16** [comment]
@dansi21 No, I wouldn't wait for this as it's most likely years until it'll be finished, if ever.
